src/services/navigationBar.ts

```diff
@@ -812,4 +828,42 @@ namespace ts.NavigationBar {
 
         return sourceFileItem.childItems;
     }
+
+    const anonFnText = "<function>";
+    const anonClassText = "<class>";
+
+    // Get the name for a (possibly anonymous) class/function expression.
+    function declarationExpressionName(node: FunctionExpression | FunctionDeclaration | ArrowFunction | ClassLikeDeclaration): string {
+        if (node.name && getFullWidth(node.name) > 0) {
+            return declarationNameToString(node.name);
+        }
+        // See if it is a var initializer. If so, use the var name.
+        else if (node.parent.kind === SyntaxKind.VariableDeclaration) {
+            return declarationNameToString((node.parent as VariableDeclaration).name);
+        }
+        // See if it is of the form "<expr> = function(){...}". If so, use the text from the left-hand side.
+        else if (node.parent.kind === SyntaxKind.BinaryExpression &&
+            (node.parent as BinaryExpression).operatorToken.kind === SyntaxKind.EqualsToken) {
+            return (node.parent as BinaryExpression).left.getText();
+        }
+        // See if it is a property assignment, and if so use the property name
+        else if (node.parent.kind === SyntaxKind.PropertyAssignment && (node.parent as PropertyAssignment).name) {
+            return (node.parent as PropertyAssignment).name.getText();
+        }
+        // Default exports are named "default"
+        else if (node.flags & NodeFlags.Default) {
+            return "default";
+        }
+        else {
+            return node.kind === SyntaxKind.ClassExpression ? anonClassText : anonFnText;
+        }
+    }
+
+    function isAnonFn(item: NavigationBarItem): boolean {
+        return item.text === anonFnText;
+    }
+
+    function isDeclarationExpression(node: Node) {

```

Comment:
DeclarationExpression is not a concept in the language. i would rather call it `isFunctionOrClassExpression`